zcash_transparent/src/bundle.rs

```diff
@@ -258,6 +270,75 @@ impl TxIn<Authorized> {
     }
 }
 
+impl TxIn<builder::Coinbase> {
+    /// Creates an input for a coinbase transaction.
+    ///
+    /// - Does not support creating inputs for the genesis block.
+    /// - The `miner_data` parameter can carry arbitrary data, which will be included in the input's
+    ///   `script_sig` field, as long as the field's length stays within
+    ///   [`coinbase::MAX_COINBASE_SCRIPT_LEN`] bytes.
+    pub fn coinbase(height: BlockHeight, miner_data: PushValue) -> Result<Self, coinbase::Error> {
```

Comment:
```suggestion
    pub fn coinbase(height: BlockHeight, miner_data: Option<PushValue>) -> Result<Self, coinbase::Error> {
```
and corresponding changes below.